internal/commands/accesslist/create_test.go

```diff
@@ -0,0 +1,218 @@
+package accesslist
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"github.com/Netflix/go-expect"
+)
+
+func TestAllowedIPCreateHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	allowedIPID := "allowedIPID"
+	allowedIPAddress := "allowedIPAddress"
+	allowedIPComment := "allowedIPComment"
+	allowedIPUseCurrent := false
+	allowedIPAllowAll := false
+	app := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+
+	t.Run("should create an allowed IP", func(t *testing.T) {
+		out, ui := mock.NewUI()
+
+		realmClient := mock.RealmClient{}
+		var capturedFilter realm.AppFilter
+		var capturedGroupID, capturedAppID, capturedIPAddress, capturedComment string
+		var capturedUseCurrent bool
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			capturedFilter = filter
+			return []realm.App{app}, nil
+		}
+
+		realmClient.AllowedIPCreateFn = func(groupID, appID, ipAddress, comment string, useCurrent bool) (realm.AllowedIP, error) {
+			capturedGroupID = groupID
+			capturedAppID = appID
+			capturedIPAddress = ipAddress
+			capturedComment = comment
+			capturedUseCurrent = useCurrent
+			return realm.AllowedIP{allowedIPID, allowedIPAddress, allowedIPComment, allowedIPUseCurrent}, nil
+		}
+
+		cmd := &CommandCreate{createInputs{
+			ProjectInputs: cli.ProjectInputs{
+				Project: projectID,
+				App:     appID,
+			},
+			Address:    allowedIPAddress,
+			Comment:    allowedIPComment,
+			UseCurrent: allowedIPUseCurrent,
+			AllowAll:   allowedIPAllowAll,
+		}}
+
+		assert.Nil(t, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+		assert.Equal(t, "Successfully created allowed IP, id: allowedIPID\n", out.String())
```

Comment:
[test/style nit] I'm personally not the biggest fan of having simple `string`/`bool` vars declared in a test for re-usability (see `allowedIPID := "allowedIPID"` above), but if we are going to have them then we should use them in the test assert via `fmt.Sprintf(...)`